pkg/cmd/admin/prune/builds.go

```diff
@@ -34,98 +35,144 @@ By default, the prune operation performs a dry run making no changes to internal
-	cmd.Flags().DurationVar(&cfg.KeepYoungerThan, "keep-younger-than", cfg.KeepYoungerThan, "Specify the minimum age of a Build for it to be considered a candidate for pruning.")
-	cmd.Flags().IntVar(&cfg.KeepComplete, "keep-complete", cfg.KeepComplete, "Per BuildConfig, specify the number of builds whose status is complete that will be preserved.")
-	cmd.Flags().IntVar(&cfg.KeepFailed, "keep-failed", cfg.KeepFailed, "Per BuildConfig, specify the number of builds whose status is failed, error, or cancelled that will be preserved.")
+	cmd.Flags().BoolVar(&opts.Confirm, "confirm", opts.Confirm, "Specify that build pruning should proceed. Defaults to false, displaying what would be deleted but not actually deleting anything.")
+	cmd.Flags().BoolVar(&opts.Orphans, "orphans", opts.Orphans, "Prune all builds whose associated BuildConfig no longer exists and whose status is complete, failed, error, or cancelled.")
+	cmd.Flags().DurationVar(&opts.KeepYoungerThan, "keep-younger-than", opts.KeepYoungerThan, "Specify the minimum age of a Build for it to be considered a candidate for pruning.")
+	cmd.Flags().IntVar(&opts.KeepComplete, "keep-complete", opts.KeepComplete, "Per BuildConfig, specify the number of builds whose status is complete that will be preserved.")
+	cmd.Flags().IntVar(&opts.KeepFailed, "keep-failed", opts.KeepFailed, "Per BuildConfig, specify the number of builds whose status is failed, error, or cancelled that will be preserved.")
 
 	return cmd
 }
+
+// Complete turns a partially defined PruneBuildsOptions into a solvent structure
+// which can be validated and used for pruning builds.
+func (o *PruneBuildsOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	if len(args) > 0 {
+		return kcmdutil.UsageError(cmd, "no arguments are allowed to this command")
+	}
+
+	o.Out = out
+
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	o.Client = osClient
+
+	buildConfigList, err := osClient.BuildConfigs(kapi.NamespaceAll).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	buildConfigs := []*buildapi.BuildConfig{}
+	for i := range buildConfigList.Items {
+		buildConfigs = append(buildConfigs, &buildConfigList.Items[i])
+	}
+
+	buildList, err := osClient.Builds(kapi.NamespaceAll).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	builds := []*buildapi.Build{}
+	for i := range buildList.Items {
+		builds = append(builds, &buildList.Items[i])
+	}
+
+	options := prune.PrunerOptions{
+		KeepYoungerThan: o.KeepYoungerThan,
+		Orphans:         o.Orphans,
+		KeepComplete:    o.KeepComplete,
+		KeepFailed:      o.KeepFailed,
+		BuildConfigs:    buildConfigs,
+		Builds:          builds,
+	}
+
+	o.Pruner = prune.NewPruner(options)
+
+	return nil
+}
+
+// Validate ensures that a PruneBuildsOptions is valid and can be used to execute pruning.
+func (o *PruneBuildsOptions) Validate() error {
+	// catch programmer error
+	if o.Pruner == nil && o.Confirm {
```

Comment:
I'm adding it here just for the sake of catching programmer error sooner, hopefully and presenting user with better error instead of nil pointer error.